core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1503,7 +1503,10 @@ class KafkaApis(val requestChannel: RequestChannel,
 
     def sendResponseCallback(producerId: Long, result: TransactionResult)(responseStatus: Map[TopicPartition, PartitionResponse]): Unit = {
       trace(s"End transaction marker append for producer id $producerId completed with status: $responseStatus")
-      val partitionErrors = responseStatus.mapValues(_.error).asJava
+      val partitionErrors = new util.HashMap[TopicPartition, Errors]()
+      responseStatus.foreach { case (topicPartition, partitionResponse) =>
+        partitionErrors.put(topicPartition, partitionResponse.error)
+      }
       val previous = errors.putIfAbsent(producerId, partitionErrors)
       if (previous != null)
         previous.putAll(partitionErrors)
```

Comment:
Yeah, I will file one. I was hoping to get confirmation whether this was indeed possible. I guess I will investigate myself.

@apurvam By the way, if we have to allow for multiple entires per producerId, then I think we will need to do more to protect access to the per-producer map. You could either use a nested concurrent map or just synchronize on the nested map. Note also that we will need to take into account the code below which also modifies this collection to handle the UNSUPPORTED_FOR_MESSAGE_FORMAT case.